skins/cat17/src/app/tests/view_handlers/test_section_info.js

```diff
@@ -422,3 +423,146 @@ test( 'Opened longtext are shut', function ( t ) {
 
 	t.end();
 } );
+
+test( 'Donor type info without entered data indicated correctly', function ( t ) {
+	var container = createContainerElement(),
+		icon = createElement(),
+		text = createElement(),
+		longText = createElement(),
+		handler = objectAssign( Object.create( SectionInfo.DonorTypeSectionInfo ), {
+			container: container,
+
+			icon: icon,
+			text: text,
+			longText: longText,
+
+			valueIconMap: { person: 'icon-person', firma: 'icon-firma', anonym: 'icon-anonym' },
+			valueTextMap: { person: 'Privatperson', firma: 'Firma', anonym: 'anonym' },
+
+			countryNames: { DE: 'Deutschland', AT: 'Österreich' }
+		} );
+
+	global.jQuery = jQueryPseudoHtmlGenerator;
+
+	icon.data.withArgs( 'display-error' ).returns( true );
+	text.data.withArgs( 'empty-text' ).returns( 'nothing entered so far' );
+
+	handler.update( 'person', '', '', '', '', '', '', '', '', 'DE', '', { dataEntered: false, isValid: null } );
+
+	t.ok( container.addClass.withArgs( 'disabled' ).calledOnce, 'no data entered reflected in style' );
+	t.ok( icon.addClass.withArgs( 'icon-error' ).calledOnce, 'icon set per address type' );
+	t.ok( text.text.withArgs( 'nothing entered so far' ).calledOnce, 'fallback address type text is set' );
+	t.equals( longText.html.args[ 0 ][ 0 ].toString(), '<span>', 'long text filled with custom mark-up' );
```

Comment:
Is this one logical assertion or several? I tried verifying it is a single one but was unable to do this without investigating the context a bunch more.

If it is not a single one, meaning this test is a `"when scenario x, scenario x results should happen"` kind of test, rather than a set of `"when scenario x, then A should happen", "when scenario x, then B should happen", "when scenario x, then C should happen"`, then I suggest creating those dedicated tests instead. Which would make extracting setup code even more important.

If it is a single one, then putting it in a dedicated function, to make it clear it is a single one, and to not mix levels of abstraction, is probably still a good idea.